src/coreclr/jit/emit.h

```diff
@@ -884,16 +884,6 @@ class emitter
         }
         void idCodeSize(unsigned sz)
         {
-            if (sz > 15)
```

Comment:
Since you are removing the workaround - does it mean we don't overestimate instruction sizes anymore?

Reply:
Double counting the REX prefix was the only place AFAIK because of which we were overcalculating the instruction size. Other overestimation still exists and I have opened #57368 to track it.